frontend/src/modules/machinery/components/Machinery.js

```diff
@@ -47,18 +52,27 @@ export default class Machinery extends React.Component<Props> {
         // Clear search field after switching to a different entity
         if (machinery.entity && !prevProps.machinery.entity) {
             this.searchInput.current.value = '';
+            this.setState({ page: 1 });
         }
     }
 
     handleResetSearch = () => {
         this.props.searchMachinery('');
+        this.setState({ page: 1 });
     };
 
     submitForm = (event: SyntheticKeyboardEvent<>) => {
         event.preventDefault();
         this.props.searchMachinery(this.searchInput.current.value);
     };
 
+    getMoreResults = () => {
+        const nextPage = this.state.page + 1;
```

Comment:
When setting state based on a current state value, please note you should do that in an updater function, [because state updates may be asynchronous](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous). So your code should read:
```js
this.setState((state) => ({
    page: state.page + 1,
}));
```

_(Please also note this may look a bit different on the surface when using hooks, e.g. `const [page, setPage] = useState(0); ... setPage(page + 1);`)_

You may ask, what about `nextPage` two lines below, then? That code is meant to go in a callback function as the second parameter to `setState()`, where you have access to the updated state (hence `this.state.page` will point to the _next_ page), although it could as well be written as part of the `componentDidUpdate` lifecycle method. [The docs recommend the latter](https://reactjs.org/docs/react-component.html#setstate), actually:
> The second parameter to `setState()` is an optional callback function that will be executed once `setState` is completed and the component is re-rendered. Generally we recommend using `componentDidUpdate()` for such logic instead.

Either way it'll be fine. The important thing is to understand the potential for asynchronous updates and how to deal with them.

Reply:
Setting the value increment on `const nextPage` and not using the state directly was my approach to handling the asynchronicity of state updates. Albeit, I wasn't convinced it was the best approach 😁 

Thanks for the resource links as they were really helpful in connecting the reasoning behind your thoughts. I'll go back over this and see if I can clean it up a bit.

Reply:
I've been unable to modify this code without raising additional issues.

If I just increment the page number in the `getMoreResults` function using the updater function for `setState` and then call `searchMachinery` in `componentDidUpdate` based on if `this.state.page !== prevState.page` I get the correct behavior for loading the additional search results. However, the functions in the `actions` file look for the existence of `page` to determine if they are to reset the overall state and this is not properly updated within this flow so the `Load More` button remains in certain instances.

If I keep the call to `searchMachinery` in the `getMoreResults` function then state has not had a chance to update and I do not get the proper value for page within the call.

Currently, I am taking the current value of `this.state.page`, adding 1, and storing that value in a variable that I then use to both update the state of `page` and call to `searchMachinery`.  This flow is successful in all scenarios that I need for this component and as such I will leave it for now.

Feel free to let me know if I am just missing something as I'm happy to revisit this if that is the case.

Reply:
If I understood correctly the problem you were facing, resetting the value of `machinery.hasMore` in the reducer as part of handling the `RESET` action seems to be the natural thing to do here. I just tried this locally with the latest state of this PR and it works fine, although I may be missing some details or relevant context.

Reply:
Yes, I had overlooked that previously and I've made that adjustment on my local branch now. 

However, the problem I encountered when calling to `searchMachinery` from `componentDidUpdate` was that after the `Load more` button has been clicked, if I attempt to reset the search by clicking the  `x` button, the call to `searchMachinery` is triggered with the reset of `page` to the starting value of 1 and the first page of results is returned along with t`hasMore` as true and so the button again shows.

Not sure if that made sense, so here is a screencast: 

https://user-images.githubusercontent.com/19564687/107890727-4469c480-6ecf-11eb-8ea1-14e5c138e7b2.mov

Reply:
Thanks for the screencast, I see much better now what problem you are running into!

> then call searchMachinery in componentDidUpdate based on if `this.state.page !== prevState.page`

This should only apply if it's not the initial page, i.e. `this.state.page !== 1`, because we don't want a `machinery.reset` action to trigger this behavior.

For the scenario you are dealing with here, I believe it's probably easier to reason about the code when using the callback to `setState`:

```js
const inputValue = this.searchInput.current.value;
this.setState(
    (state) => ({ page: state.page + 1 }),
    () => {
        this.props.searchMachinery(inputValue, this.state.page);
    }
);
```

I couldn't hit any of the bugs you mentioned with this code above (nor with the `this.state.page !== 1` check in `componentDidUpdate`).